packages/dds/matrix/src/test/utils.ts

```diff
@@ -222,3 +224,25 @@ export function insertFragmented(
 
 	return matrix;
 }
+
+/**
+ * Creates a local matrix with the specified size and initial value.
+ */
+export function createLocalMatrix(
+	id: string,
+	size: number,
+	initialValue: string,
+): ISharedMatrix & IChannel {
+	const matrix = matrixFactory.create(new MockFluidDataStoreRuntime(), id);
+	for (let i = 0; i < size; i++) {
+		matrix.insertRows(i, 1);
+		matrix.insertCols(i, 1);
+	}
+
+	for (let row = 0; row < size; row++) {
```

Comment:
Setting all of the values up-front satisfies the "dense" case called out in the task description. We will also want to add "sparse" cases (where the values are not filled in initially). Totally fine to do this in a follow-up PR though. Small PRs are nice :)

Reply:
But it's probably worth leaving a TODO note somewhere that we'll want to cover the sparse case as well.

Reply:
Yeah, that sounds like a good way to do it. You'd just want to make sure that the function is documented clearly so users know how `undefined` affects initialization.